maker/src/cfd.rs

```diff
@@ -504,41 +507,60 @@ impl<O, T, W> Actor<O, T, W> {
     async fn handle_accept_settlement(&mut self, msg: AcceptSettlement) -> Result<()> {
         let AcceptSettlement { order_id } = msg;
 
-        match self
-            .settlement_actors
-            .send_async(&order_id, collab_settlement::Accepted)
-            .await
-        {
-            Ok(_) => Ok(()),
-            Err(NotConnected(e)) => {
-                self.executor
-                    .execute(order_id, |cfd| {
-                        Ok(cfd.fail_collaborative_settlement(anyhow!(e)))
-                    })
-                    .await?;
-
-                bail!("Accept failed: No settlement in progress for order {order_id}")
+        // TODO: Handle case when taker upgraded after opening and *can* use libp2p
+        let cfd = self.db.load_open_cfd::<Cfd>(order_id, ()).await?;
+        if can_use_libp2p(&cfd) {
+            self.libp2p_collab_settlement
+                .send(daemon::collab_settlement::maker::Accept { order_id })
+                .await
+                .map_err(|e| anyhow!(e))?
+        } else {
+            match self
+                .settlement_actors
+                .send_async(&order_id, collab_settlement::Accepted)
+                .await
+            {
+                Ok(_) => Ok(()),
+                Err(NotConnected(e)) => {
+                    self.executor
+                        .execute(order_id, |cfd| {
+                            Ok(cfd.fail_collaborative_settlement(anyhow!(e)))
+                        })
+                        .await?;
+
+                    bail!("Accept failed: No settlement in progress for order {order_id}")
+                }
             }
         }
     }
 
     async fn handle_reject_settlement(&mut self, msg: RejectSettlement) -> Result<()> {
         let RejectSettlement { order_id } = msg;
 
-        match self
-            .settlement_actors
-            .send_async(&order_id, collab_settlement::Rejected)
-            .await
-        {
-            Ok(_) => Ok(()),
-            Err(NotConnected(e)) => {
-                self.executor
-                    .execute(order_id, |cfd| {
-                        Ok(cfd.fail_collaborative_settlement(anyhow!(e)))
-                    })
-                    .await?;
-
-                bail!("Reject failed: No settlement in progress for order {order_id}")
+        // TODO: Handle case when taker upgraded after opening and *can* use libp2p
```

Comment:
I think this still has to be handled before we can merge, otherwise takers cannot collab close previously opened positions.

Reply:
Yes, that's described in the PR description. There is a ticket for it in this iteration: #2156